satpy/writers/cf_writer.py

```diff
@@ -499,14 +514,20 @@ def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compr
                 If True, flatten dict-type attributes
             exclude_attrs (list):
                 List of dataset attributes to be excluded
-
+            valid_cf_dataset_name (bool):
+                Prepend dataset name with CHANNEL_ if staring with a digit
         """
         if exclude_attrs is None:
             exclude_attrs = []
 
         new_data = dataarray.copy()
         if 'name' in new_data.attrs:
             name = new_data.attrs.pop('name')
+            if valid_cf_dataset_name and name[0].isdigit():
+                _orig_name = name
+                name = 'CHANNEL_' + name
```

Comment:
What about renaming this keyword argument `numeric_name_prefix` and if it is None, don't do anything. If it is specified then it is assumed to be a string. So someone could do `scn.save_datasets(writer='cf', numeric_name_prefix='ch')` if they wanted to?

Reply:
I second that.

Reply:
Sounds like a good idea.

I just have not figured out how to read this back if we are to skip the extra attribute I have added for now.